dozer/cogs/filter.py

```diff
@@ -138,7 +138,34 @@ def load_filters(self, guild_id):
         await ctx.send(embed=embed)
         self.load_filters(ctx.guild.id)
 
-    add.example_usage = "`{prefix}filter whitelist add Swear` - Makes it so that \"Swear\" will be filtered"
+    add.example_usage = "`{prefix}filter add Swear` - Makes it so that \"Swear\" will be filtered"
+
+    @guild_only()
+    @has_permissions(manage_guild=True)
+    @filter.command()
+    async def edit(self, ctx, filter_id, pattern):
+        """Edit an already existing filter using a new pattern. A filter's friendly name cannot be edited."""
+        try:
+            re.compile(pattern)
+        except re.error as err:
+            await ctx.send("Invalid RegEx! ```{}```".format(err.msg))
+            return
+        with db.Session() as session:
+            result = session.query(WordFilter).filter_by(id=filter_id, guild_id=ctx.guild.id).one_or_none()
+            old_pattern = result.pattern
+            if result is None:
+                await ctx.send("That filter ID does not exist or does not belong to this guild.")
+                return
+            result.pattern = pattern
```

Comment:
That's partially the reason that you can't change the friendly name for the filter, as the actual content behind the filter shouldn't be changed from one word to another, it should just be fixing the filter slightly (as we've done several times on the FRC server at this point)

Reply:
That almost seems worse, as what `&filter` shows and what the filters actually do can be totally separate. `&filter true` will show what's really being filtered, but only if the user understands regex.

Reply:
Just because we implemented an edit command doesn't change that though. You can set a friendly name in the first place that completely is different than the pattern from the initial create command. I'm not entirely sure what your third sentence is trying to say, but if you're trying to make the argument that just because all the prior edits have been creating a new filter, I don't think that's a valid excuse to change this. I don't think it's very intuitive (and frustrating) to have to make a entirely new filter for a simple change in a pattern.